toolbox/_common.py

```diff
@@ -0,0 +1,110 @@
+import itertools
+import subprocess
+import os
+import time
+import sys
+import shlex
+from pathlib import Path
+
+
+top_dir = Path(__file__).resolve().parent.parent
+
+
+class PlaybookRun:
+    """
+    Playbook runner
+
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    If you're seeing this text, put the --help flag earlier in your list
+    of command-line arguments, this is a limitation of the CLI parsing library
+    used by the toolbox.
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+    """
+    def __init__(self, playbook_name, opts: dict = None):
+        if opts is None:
+            opts = {}
+
+        self.playbook_name = playbook_name
+        self.opts = opts
+
+    def __str__(self):
+        return ""
+
+    def _run(self):
+        run_ansible_playbook(self.playbook_name, self.opts)
+
+
+def flatten(lst):
+    return itertools.chain(*lst)
+
+
+def run_ansible_playbook(playbook_name, opts: dict = dict()):
+    version_override = os.environ.get("OCP_VERSION")
+    if version_override is not None:
+        opts["openshift_release"] = version_override
+
+    if os.environ.get("ARTIFACT_DIR") is None:
+        os.environ["ARTIFACT_DIR"] = f"/tmp/ci-artifacts_{time.strftime('%Y%m%d')}"
+        print(f"Using '{os.environ['ARTIFACT_DIR']}' to store the test artifacts (default value for ARTIFACT_DIR).")
+    else:
+        print(f"Using '{os.environ['ARTIFACT_DIR']}' to store the test artifacts.")
+    opts["artifact_dir"] = os.environ["ARTIFACT_DIR"]
+
+    os.environ["ARTIFACT_DIRNAME"] = '__'.join(sys.argv[1:3])
+
+    artifact_dir = Path(os.environ["ARTIFACT_DIR"])
+    artifact_dir.mkdir(parents=True, exist_ok=True)
+
+    previous_extra_count = len(list(artifact_dir.glob("*__*")))
+    os.environ["ARTIFACT_EXTRA_LOGS_DIR"] = str(
+            Path(os.environ["ARTIFACT_DIR"]) /
+            f"{previous_extra_count}__{os.environ['ARTIFACT_DIRNAME']}"
```

Comment:
could you change `f"{previous_extra_count}` to `f"{previous_extra_count:03d}` to force the number to be printed with 3 digits?

Reply:
yeah that bothered me and I forgot about it